src/main/java/hudson/plugins/filesystem_scm/FolderDiff.java

```diff
@@ -192,70 +213,76 @@ private AndFileFilter createAntPatternFileFilter() {
-
-        // now get the list of all sources in workspace (destination)
-        Iterator<File> it = (Iterator<File>) FileUtils.iterateFiles(dst, TrueFileFilter.TRUE, TrueFileFilter.TRUE);
+        Collection<File> allSources = new ArrayList<>();
+        if (src.isDirectory()) {
+            allSources = (Collection<File>) FileUtils.listFiles(src, fileFilter, dirFilter);
+        }
 
         ArrayList<Entry> list = new ArrayList<Entry>();
-        while (it.hasNext()) {
-            File file = it.next();
-            try {
-                String relativeName = getRelativeName(file.getAbsolutePath(), dst.getAbsolutePath());
-                File tmp = new File(src, relativeName);
-                if (!allSources.contains(tmp) && (null == allowDeleteList || allowDeleteList.contains(relativeName))) {
-                    log("Deleted file: " + relativeName);
-                    list.add(new Entry(relativeName, Entry.Type.DELETED));
-                    if (breakOnceFound)
-                        return list;
-                    if (!testRun) {
+        if (dst.isDirectory()) {
+            // now get the list of all sources in workspace (destination)
+            Iterator<File> it = (Iterator<File>) FileUtils.iterateFiles(dst, TrueFileFilter.TRUE, TrueFileFilter.TRUE);
+            while (it.hasNext()) {
+                File file = it.next();
+                try {
+                    String relativeName = getRelativeName(file.getAbsolutePath(), dst.getAbsolutePath());
+                    File tmp = new File(src, relativeName);
+                    if (!allSources.contains(tmp)
+                            && (null == allowDeleteList || allowDeleteList.contains(relativeName))) {
+                        addAndLog(list, relativeName, Type.DELETED);
+                        if (breakOnceFound) {
+                            return list;
+                        }
                         try {
-                            boolean deleted = file.delete();
+                            boolean deleted = deleteFile(file);
                             if (!deleted) {
                                 log("file.delete() failed: " + file.getAbsolutePath());
                             }
                         } catch (SecurityException e) {
                             log("Can't delete " + file.getAbsolutePath(), e);
                         }
                     }
+                } catch (IOException e) {
+                    log(e);
                 }
-            } catch (IOException e) {
-                log(e);
             }
         }
         return list;
     }
 
+    private IOFileFilter getDirFilter() {
+        return ignoreHidden ? HiddenFileFilter.VISIBLE : TrueFileFilter.TRUE;
+    }
+
+    /**
+     * should delete the given file
+     * 
+     * @param file
+     *            the file to delete
+     * @return true if successful
+     */
+    protected boolean deleteFile(File file) {
+        return file.delete();

```

Comment:
Better to use Java 7's `java.nio.Files` API with better error propagation (IOException)